conf/i18n/translatehelpervisitor.js

```diff
@@ -0,0 +1,163 @@
+const Handlebars = require('handlebars');
+const TranslationPlaceholder = require('./models/translationplaceholder');
+
+/**
+ * TranslateHelperVisitor accepts a handlebars AST, and replaces all translate placeholders
+ * with either static translations, or a runtime translation helper.
+ */
+class TranslateHelperVisitor {
+  constructor (translationResolver) {
+    this._visitor = new Handlebars.Visitor();
+    // This line puts the Handlebars.Visitor instance into mutation mode.
+    // https://github.com/handlebars-lang/handlebars.js/blob/master/docs/compiler-api.md#ast-visitor.
+    this._visitor.mutating = true;
+    this._visitor.MustacheStatement = this._MustacheStatement.bind(this);
+
+    this._translationResolver = translationResolver;
+    this._validHelpers = ['translate'];
+    this._runtimeTranslationHelper = 'runtimeTranslation';
+    this._contextParam = 'context';
+    this._pluralFormParam = 'pluralForm';
+  }
+
+  /**
+   * Dispatches TranslateHelperVisitor on a Handlebars AST.
+   * @param {hbs.AST.Program} ast
+   */
+  accept (ast) {
+    this._visitor.accept(ast);
+  }
+
+  /**
+   * _MustacheStatement() is automatically called on any MustacheStatement nodes by the visitor.
+   * Returning undefined leaves the node unaffected, otherwise it replaces it with the
+   * returned value.
+   * @param {hbs.AST.MustacheStatement} statement
+   * @returns {hbs.AST.MustacheStatment|undefined}
+   */
+  _MustacheStatement (statement) {
+    if (!this._validHelpers.includes(statement.path.original)) {
```

Comment:
Lets do some renaming here to make it clearer what's happening. We're looking to see if the statement's "path" corresponds to a translate helper

Reply:
added a variable!